### ISSUE TITLE
[Bug]: page.route stops working after awhile for a few devs

### ISSUE THREAD

**Issue by mentaman** [microsoft/playwright#7941]
### Playwright version

1.10.0

### Operating system

MacOS

### What browsers are you seeing the problem on?

Chromium

### Other information

We're all using
Nodejs 12.17.0
mac - ProductVersion:	10.15.7

Works in mine but at on theirs computers it doesn't work well

### What happened? / Describe the bug

it stops getting into page.route for some after a few seconds they use my playwright software.
```
try {
     await page.route(`${remote_url}**\/*.{js}`, async (route) => {
                try {
                    const url: string = route.request().url();
                    console.log(`overriding ${url)`)
                    const override = await this.getOverrideRouteForUrl(url);
                    if(override) {
                        return route.fulfill(override);
                    }
                    return route.continue();
              } catch(e) { console.error("inside route error; "); console.error(e); }
    }
} catch(e) {
    console.error("outside route error; "); console.error(e);
}
```

I can show the code in getOverrideRouteForUrl but it's a little long, in general it sends an axios message and gets the data from it. the axios has a timeout.

That's the code in general, it's kind of a proxy that gets webpack from local and override the production build,
this way we can debug production and change code easily.
I might rewrite it as an opensource soon and share the code, haven't got the time yet.

For some weird reason, it doesn't work well for a few developers.
It works at the beginning but after awhile it stops showing the message
                    console.log(`overriding ${url)`);
it's always reproduced for them, quite at the start. 
and no longer override the requests for them.
but the browser works well other than that.
never reproduced to me. 

and it never gets to the catch(error), so no information about why it stopped.

any idea why it stops? are there maybe more logs to add about why it stops?
I have no directions at all.
two of them said they have other crashes in their computer related to not having a lot of free ram, we tried playing with the nodejs --max-old-space-size=10240 and close stuff in their computer

**Comment by mxschmitt** [comment]
Since you are using version `1.10.0` I would recommend to upgrade to the latest one  which is currently `1.13.1`.

I wild guess would be that the Promise for `getOverrideRouteForUrl ` does not resolve and thats why it stalls. Would be great if you could provide us debug logs via setting the `DEBUG=pw:api` env var, there we also see the API calls and which ones did _maybe_ stall on our side.

**Comment by mentaman** [comment]
@mxschmitt Aside of axios everything is sync inside,
And the axios has a 10 seconds timeout.

If one request didn't resolve yet it would effect all the other requests too? 

Also the browser keeps working after it and receiving requests, it just doesn't go through the page.route anymore once this happens, does it make sense that it would have this behaviour if it didn't resolve? 
the log at the beginning of the page.route is not shown anymore although the requests get resolved after this bug starts.

It's the exact same code in our pcs and theirs, and in most developers pcs it works, just for a few it doesn't.


Awesome will try the DEBUG=pw:api and see what it shows thanks!

**Comment by mentaman** [comment]
hey @mxschmitt managed to reproduce on my computer today, haven't done anything special, not sure why it's reproduced today in my computer too.

tried your suggestion. and I'm on the version. 1.13.1

this is the log:

 pw:api   navigated to "http://myweb.com/test-proxy.js" +43ms
  pw:api   "domcontentloaded" event fired +77ms
  pw:api   "load" event fired +5ms
  pw:api   "networkidle" event fired +499ms

this is what I see with the debug=pw:api ^

for test-proxy.js I have only this code:

     if(url === `${remote_url}test-proxy.js`) {
                console.log("fullfill")
                return ({
                    status: 200,
                    contentType: "text/plain",
                    body: "proxy works"
                })
            }

and it doesn't get to the "fulfill" log

**Comment by mentaman** [comment]
Okay I think the reason it was reproduced to me suddenly, might have been related to upgrading to 1.13.1 and some breaking changes maybe?
I've made a few changes and it seems to work well now to me, will see if it helps also to those who had problems before. 

What I changed aside of upgrading to try and fix it:
- changed route from `${remote_url}**\/*.${js}` to `${remote_url}**\/*.js`
- used built typescript instead of using ts-node
- removed max-old-space-size from node configuration

After these can't reproduce it anymore.

Will update if it maybe fixed it for them too, if not will ask them to run with the debug=pw:api too
